databuilder/databuilder/task/search/document_mappings.py

```diff
@@ -0,0 +1,84 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict
+from elasticsearch_dsl import Document, Text, Keyword, RankFeatures, Date, tokenizer, token_filter, Long
+from elasticsearch_dsl.analysis import analyzer
+
+general_tokenizer = tokenizer("general_tokenizer",
+                              type="char_group",
+                              tokenize_on_chars=[
+                                  "whitespace",
+                                  "_",
+                                  "-",
+                                  "."
+                              ])
+
+stemming_analyzer = analyzer("stemming_analyzer",
+                             tokenizer=general_tokenizer,
+                             filter=["lowercase", "kstem"])
+
+english_stop = token_filter("english_stop", type="stop", stopwords="_english_")
+# might not need the keywords one
+english_keywords = token_filter("english_keywords", type="keyword_marker", keywords=["example"])
+english_stemmer = token_filter("english_stemmer", type="stemmer", language="english")
+english_possessive_stemmer = token_filter("english_possessive_stemmer", type="stemmer", language="possessive_english")
+
+english_analyzer = analyzer("english_analyzer",
+                            tokenizer=tokenizer("standard_tokenizer", type="standard"),
+                            filter=[
+                                english_possessive_stemmer,
+                                "lowercase",
+                                english_stop,
+                                english_keywords,
+                                english_stemmer
+                            ])
+
+
+class SearchableResource(Document):
+    key = Keyword(required=True)
+    name = Keyword(required=True, analyzer=stemming_analyzer)
+    description = Text(analyzer=english_analyzer)  # TODO add custom analyzer
+    badges = Keyword(multi=True)
+    tags = Keyword(multi=True)
+    usage_metrics = RankFeatures(type=Long)
+    last_updated_timestamp = Date()
+
+
+class Table(SearchableResource):
+    display_name = Keyword(required=True, analyzer=stemming_analyzer)
+    database = Keyword(required=True)
+    cluster = Keyword()
```

Comment:
is there a reason cluster is not a basic `SearchableResource` field? it's common for every resource we have afaik

Reply:
features and users do not have clusters